src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -631,6 +631,10 @@ def get_msal_token(self, scopes, data):
         authority = posixpath.join(self.cli_ctx.cloud.endpoints.active_directory, tenant)
         app = ClientApplication(_CLIENT_ID, authority=authority)
         result = app.acquire_token_by_refresh_token(refresh_token, scopes, data=data)
+
+        if 'error' in result:
```

Comment:
Is this a reliable way to assert the error?

Reply:
Detecting `error` key in the `result` `dict` _is_ the reliable way to assert the error. It is MSAL's designed API:

https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/78e9ccfab7c16154ba38c0de78c495ba2ee58849/msal/application.py#L761-L766

```py
    def acquire_token_silent_with_error(
            self,
            ...):
        """Acquire an access token for given account, without user interaction.
        ...
        :return:
            - A dict containing no "error" key,
              and typically contains an "access_token" key,
              if cache lookup succeeded.
            - None when there is simply no token in the cache.
            - A dict containing an "error" key, when token refresh failed.
        """
```